src/box/lua/load_cfg.lua

```diff
@@ -615,6 +615,9 @@ local function reload_cfg(oldcfg, cfg)
             rawset(oldcfg, key, val)
             if not pcall(dynamic_cfg[key]) then
                 rawset(oldcfg, key, oldval) -- revert the old value
+                if key == "listen" then
+                    pcall(dynamic_cfg[key])
```

Comment:
Let's introduce a new table of callbacks for reverts to keep this function generic:

```lua
local dynamic_cfg_revert = {
    listen                  = private.cfg_set_listen,
}
```